th/Categorifier/TH.hs

```diff
@@ -37,14 +40,30 @@ import Data.Map (Map)
 import qualified Data.Map as Map
 import Data.Maybe (catMaybes, fromMaybe)
 import Data.These (These (..))
-import Language.Haskell.TH hiding (TyVarBndr, conP, reifyType)
+import Language.Haskell.TH hiding (TyVarBndr (..), conP, reifyType)
 import qualified Language.Haskell.TH as TH
 import PyF (fmt)
 
 #if MIN_VERSION_template_haskell(2, 17, 0)
 type TyVarBndr flag = TH.TyVarBndr flag
+
+pattern PlainTV :: TH.Name -> flag -> TyVarBndr flag
+pattern PlainTV n flag = TH.PlainTV n flag
+
+pattern KindedTV :: TH.Name -> flag -> TH.Kind -> TyVarBndr flag
+pattern KindedTV n flag k = TH.KindedTV n flag k
+
+{-# COMPLETE PlainTV, KindedTV #-}
```

Comment:
Can we pull the declarations and the `complete` pragma out of the conditionalizations?